Projects/Social-Trading/SA/Function-Libraries/SocialEntitiesProfile.js

```diff
@@ -473,6 +497,71 @@ exports.newSocialTradingFunctionLibrariesSocialEntitiesProfile = function () {
 
                 SA.projects.network.globals.memory.maps.USER_PROFILES_BY_ID.set(userProfile.id, inMemoryUserProfile)
             }
+            
+            function saveApiAppFile() {
+              let filePath = global.env.PATH_TO_SECRETS + '/';
+              let fileName = "ApisSecrets.json";
+              let fileContent;
+            
+              // Check if the folder exists. If not, create it.
+              SA.projects.foundations.utilities.filesAndDirectories.createNewDir(filePath);
```

Comment:
You should check if the file exists instead of relying on the try/catch block to error out inside of createNewDir().
if (!fs.existsSync(filePath) {
SA.projects.foundations.utilities.filesAndDirectories.createNewDir(filePath);
}